src/pages/index.js

```diff
@@ -176,21 +226,46 @@ function getCardElement(data) {
   const cardNameEl = cardElement.querySelector(".card__title");
   const cardImageEl = cardElement.querySelector(".card__image");
   const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardIsLiked = "card__like-button_liked";
   const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
 
   cardNameEl.textContent = data.name;
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
 
-  cardLikeBtn.addEventListener("click", () => {
-    cardLikeBtn.classList.toggle("card__like-button_liked");
-  });
+  // card like button -------------------------
 
-  cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
+  const isLikedByUser = Array.isArray(data.isLiked)
+    ? data.likes.some((user) => user._id === currentUserId)
+    : false;
+  if (isLikedByUser) {
+    cardLikeBtn.classList.add(cardIsLiked);
+  }
+
+  cardLikeBtn.addEventListener("click", () => {
+    const isLiked = cardLikeBtn.classList.contains(cardIsLiked);
+
+    const likeAction = isLiked
+      ? api.removeLike(data._id)
+      : api.addLike(data._id);
+
+    likeAction
+      .then((updatedCard) => {
+        const likesArray = Array.isArray(updatedCard.isLiked)
+          ? updatedCard.likes
+          : [];
+
+        const likedByUser = likesArray.some(
+          (user) => user._id === currentUserId
+        );
+        cardLikeBtn.classList.toggle(cardIsLiked, likedByUser);
+      })
+      .catch((err) => {
```

Comment:
* Likes don't work

`likes` don't exist in the `updatedCard`. There is `isLiked` for that


```js
    likeAction
      .then((updatedCard) => {
        cardLikeBtn.classList.toggle(cardIsLiked, updatedCard.isLiked);
      })
      .catch((err) => {
```